src/universal/modules/outcomeCard/components/AddSoftTeamMember.js

```diff
@@ -0,0 +1,225 @@
+import InviteTeamMembersMutation from 'universal/mutations/InviteTeamMembersMutation';
+import withAtmosphere from 'universal/decorators/withAtmosphere/withAtmosphere';
+import {connect} from 'react-redux';
+import withMutationProps from 'universal/utils/relay/withMutationProps';
+import UpdateProjectMutation from 'universal/mutations/UpdateProjectMutation';
+import {emailRegex} from 'universal/validation/regex';
+import legitify from 'universal/validation/legitify';
+import {createFragmentContainer} from 'react-relay';
+import ErrorMessageInMenu from 'universal/components/ErrorMessageInMenu';
+
+const makeValidationSchema = (allAssignees) => {
+  return legitify({
+    inviteeEmail: (value) => value
+      .trim()
+      .required('You should enter an email here')
+      .matches(emailRegex, 'That doesn’t look like an email address')
+      .test((inviteeEmail) => {
+        const alreadyInList = allAssignees.find(({email}) => email === inviteeEmail);
+        return alreadyInList && 'That person is already in the list';
+      })
+  });
+};
+
+const validateEmailAddress = (inviteeEmail, assignees, onError) => {
+  const schema = makeValidationSchema(assignees);
+  const {errors} = schema({inviteeEmail});
+  if (Object.keys(errors).length) {
+    onError(errors.inviteeEmail);
+    return false;
+  }
+  return true;
+};
+
+class AddSoftTeamMember extends Component {
+  static propTypes = {
+    area: PropTypes.string.isRequired,
+    assignRef: PropTypes.instanceOf(Element),
+    atmosphere: PropTypes.object.isRequired,
+    dirty: PropTypes.bool.isRequired,
+    setDirty: PropTypes.func.isRequired,
+    closePortal: PropTypes.func.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    isActive: PropTypes.bool.isRequired,
+    menuRef: PropTypes.instanceOf(Element),
+    projectId: PropTypes.string.isRequired,
+    error: PropTypes.any,
+    submitting: PropTypes.bool,
+    submitMutation: PropTypes.func.isRequired,
+    onCompleted: PropTypes.func.isRequired,
+    onError: PropTypes.func.isRequired,
+    styles: PropTypes.object.isRequired,
+    team: PropTypes.object.isRequired
+  };
+
+  state = {inviteeEmail: ''};
+  onSubmit = (e) => {
+    e.preventDefault();
+    const {inviteeEmail} = this.state;
+    const {
+      area,
+      assignRef,
+      atmosphere,
+      closePortal,
+      setDirty,
+      dispatch,
+      projectId,
+      submitting,
+      submitMutation,
+      onCompleted,
+      onError,
+      team
+    } = this.props;
+    if (submitting) return;
+
+    // validate
+    setDirty();
+    const {teamMembers, softTeamMembers, teamId} = team;
+    const assignees = teamMembers.concat(softTeamMembers);
+    const isValid = validateEmailAddress(inviteeEmail, assignees, onError);
+    if (!isValid) return;
+
+    // resolve
+    closePortal();
+    assignRef.focus();
+    const invitees = [{email: inviteeEmail}];
+    submitMutation();
+    const handleCompleted = (res) => {
```

Comment:
chaining 2 distinct mutations. we can't create a single "mutation" that has 2 mutation operations because we need the output of the first to feed the input of the 2nd. creating an entirely new mutation to save a roundtrip seems wasteful, especially because we use optimistic updates, so it feels instant to the client.

Reply:
Totes fine